spec/abci++/abci++_tmint_expected_behavior_002_draft.md

```diff
@@ -145,33 +141,35 @@ Let us now examine the grammar line by line, providing further details.
 >consensus-exec      = (inf)consensus-height
 >```
 
-* A consensus height consists of zero or more rounds before deciding via a call to `FinalizeBlock`.
-  In each round, the sequence of method calls depends on whether the local process is the proposer or not.
+* A consensus height consists of zero or more rounds before deciding via a call to `FinalizeBlock`,
+  followed by a call to `Commit`. If the height has zero rounds, this means the process is
+  replaying an already decided value (catch-up mode). In each round, the sequence of method calls
+  depends on whether the local process is the proposer or not.
 
 >```abnf
->consensus-height    = *consensus-round decide
+>consensus-height    = *consensus-round decide commit
 >consensus-round     = proposer / non-proposer
 >```
 
-* If the local process is the proposer of the current round, Tendermint starts by calling `PrepareProposal`.
-  No calls to methods related to vote extensions (`ExtendVote`, `VerifyVoteExtension`) can be called
-  in the present round before `PrepareProposal`. Once `PrepareProposal` is called, calls to
-  `ExtendVote` and `VerifyVoteExtension` can come in any order, although the former will be called
-  at most once in this round.
+* If the local process is the proposer of the current round, Tendermint starts by calling
+  `PrepareProposal`, followed by `ProcessProposal`. Then, optionally, the Application is asked
+  to extend its vote for that round. Calls to `VerifyVoteExtension` can come at any time: the
+  local process may be slightly late in the current round, or votes may come from a future round
+  of this height.
 
 >```abnf
->proposer            = prepare-proposal extend-proposer
->extend-proposer     = *got-vote [extend-vote] *got-vote
+>proposer            = *got-vote prepare-proposal *got-vote process-proposal [extend]
+>extend              = *got-vote extend-vote *got-vote
 >```
 
 * If the local process is not the proposer of the current round, Tendermint will call `ProcessProposal`
-  at most once. At most one call to `ExtendVote` can occur only after `ProcessProposal` is called.
+  at most once. At most one call to `ExtendVote` may occur only after `ProcessProposal` is called.
```

Comment:
Maybe specify that this sentence is describing the scenario _per round_, not for the overall execution?

Reply:
Good point. Will do